src/main/kotlin/org/wycliffeassociates/otter/common/domain/content/RecordTake.kt

```diff
@@ -29,142 +23,139 @@ class RecordTake(
         NO_AUDIO
     }
 
-    private fun getContentCount(collection: Collection, filter: (Content) -> Boolean): Single<Int> = contentRepository
-        .getByCollection(collection)
-        .map { retrieved -> retrieved.filter(filter).size }
+    private fun getContentCount(chapter: Chapter): Single<Long> = chapter.chunks.count()
 
-    private fun getNumberOfSubcollections(collection: Collection): Single<Int> = collectionRepository
-        .getChildren(collection)
-        .map { it.size }
+    private fun getNumberOfSubcollections(project: Book): Single<Long> = project.chapters.count()
 
-    private fun getMaxTakeNumber(content: Content): Single<Int> = takeRepository
-        .getByContent(content)
-        .map { takes ->
-            takes.maxBy { it.number }?.number ?: 0
-        }
+    private fun getNewTakeNumber(recordable: Recordable): Single<Int> = recordable.audio.takes.let { relay ->
+        Single.just (
+            relay.getValues(arrayOfNulls<Take>(relay.values.size))
+                .maxBy { it.number }
+                ?.number
+                ?.plus(1)
+                ?: 1
+        )
+    }
```

Comment:
This is untested, but might be better:
```
    private fun getNewTakeNumber(recordable: Recordable): Single<Int> =
        recordable.audio.takes
            .map(Take::number)
            .reduce(0, Math::max)
            .map { it + 1 }
```

Reply:
Not sure why, but this gets hung up during execution just like recordable.audio.takes.count() when I tried that

Reply:
Ohh!  It's because the relay never sends Complete.  There might be a better way of handling this, but this occurs to me: Replace `recordable.audio.takes` with `recordable.audio.takes.take(recordable.audio.takes.size)`

Reply:
That's not great either, though, is it? I wish there were a slicker way to get the buffered values from a relay, or at least its buffer size. What you had originally may be the best option.

Reply:
Maybe this is better:
```
    private fun getNewTakeNumber(recordable: Recordable): Single<Int> =
        recordable.getAllTakesObservable()
            .map(Take::number)
            .reduce(0, Math::max)
            .map { it + 1 }
```

and in Recordable,

```
    fun getAllTakesObservable(): Observable<Take> = audio.takes.take(audio.takes.values.size.toLong())
```

`getNewTakeNumber()` probably shouldn't depend on the `takes` relay being specifically a ReplayRelay anyway

Reply:
Or maybe we should really create an interface with a method to get all takes? Then either Recordable (or better yet, AssociatedAudio?) can implement this interface, and `getNewTakeNumber()` would take an object that implements the interface. @sarabiaj does this sound good?
```
class AssociatedAudio: I[SomeInterface] {
...
}

interface I[SomeInterface] {
    fun getAllTakesObservable()
    fun acceptTake()
}
```
Or perhaps we don't need this interface and we can just depend on AssociatedAudio- I might be taking the Open-Closed principle too far. I'm definitely still learning here

Reply:
I'd like us to add an extension method, `ReplayRelay.takeBuffered() = this.take(this.values.count)`.

I opened an [issue on RxRelay](https://github.com/JakeWharton/RxRelay/issues/52), and he can implement it in a more efficient way, but we should do it as an extension method in the meantime.

Then we can make the domain code do whatever.  Extension methods or interfaces as you're describing should be fine, but just making the original `RecordTake::getNewTakeNumber` use the new relay interface would feel pretty good to me.

What do you think?

Reply:
Yeah I think that's a good idea!

Reply:
Haha, ok let's _ctrl-z_ then! Thanks for looking into all of this anyway, I feel like the change-request had some potential and was a good idea.

Anyway, perhaps instead of `AssociatedAudio::getAllTakesObservable`, I will create an `AssociatedAudio::getAllTakes` method that simply returns and array of all the takes (and only calls `getValues` once, as you mentioned!)